src/core/worker.js

```diff
@@ -956,18 +956,23 @@ var WorkerMessageHandler = {
       docParams = null; // we don't need docParams anymore -- saving memory.
     });
     return workerHandlerName;
-  }
+  },
+  initializeFromPort(port) {
+    var handler = new MessageHandler('worker', 'main', self);
```

Comment:
`self` should be `port` here and below, right? Otherwise `port` is unused and you pass `self` as parameter to this method below.